src/main/java/com/hazelcast/aws/AwsDiscoveryStrategy.java

```diff
@@ -57,6 +56,8 @@
     private static final Integer DEFAULT_CONNECTION_RETRIES = 3;
     private static final int DEFAULT_CONNECTION_TIMEOUT_SECONDS = 10;
     private static final String DEFAULT_HOST_HEADER = "ec2.amazonaws.com";
+    private static final String AWS_REGION_REGEX = "\\w{2}-\\w+-\\d(?!.+)";
```

Comment:
Have you tested this regex against all AWS Regions? Like e.g. `us-gov-east-1`?

You could actually make a test with the names of all current AWS regions to make sure we don't miss anything. To do it, you could make `validateRegion()` package-private with the comment that it's used in the tests and add the dedicated test for that. I know it's not nice to break the encapsulation because of tests, but we'll refactor `hazelcast-aws` at some point, and I'd value the correctness over design here.

Reply:
Yeah, sorry about that, I missed this one when I looked at the list. I'll fix this.

About making it package-private because of tests, my main concern was exactly encapsulation. But if it's alright to do it, I'll fix it.

Reply:
Quick question here: since I'll test on every region, I was thinking about creating a test with `@Parameters`, using `@RunWith(Parameterized.class)` as the runner, since it's jUnit4, but we're already using the `@RunWith(HazelcastParallelClassRunner.class)`.

So, which do you think would be the best solution:
- Creating a separate class and using `@RunWith(Parameterized.class)` to use `@Parameters`;
- Loop the values and assert each one inside of it;
- Another suggestion.

Reply:
For this simple case I'd just do like:

```
public void validateRegion() {
    validateRegion("us-east-2");
    validateRegion("us-east-1");
    ...
}
```

And a separate test with one case that is not a valid region. Maybe you don't even need to test ALL the regions, but at least the ones that looks different. Looking at [this list](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html), I'l put just one from each area, like: `us-east-2`, `us-west-1`, `ap-east-1`, `ap-south-1`, `ap-northeast-3`, `ap-southeast-1`, `ca-central-1`, ... .

Reply:
I tried validating more thoroughly, checking for specific cardinal directions used by AWS, and separated two tests to make it a little clearer.